app/src/main/java/com/github/libretube/api/poToken/JavaScriptUtil.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.libretube.api.poToken
+
+import okio.ByteString.Companion.decodeBase64
+import okio.ByteString.Companion.toByteString
+
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonPrimitive
+import kotlinx.serialization.json.long
+
+/**
+ * Parses the raw challenge data obtained from the Create endpoint and returns an object that can be
+ * embedded in a JavaScript snippet.
+ */
+fun parseChallengeData(rawChallengeData: String): String {
+    val scrambled = Json.parseToJsonElement(rawChallengeData).jsonArray
+
+    val challengeData = if (scrambled.size > 1 && scrambled[1].jsonPrimitive.isString) {
```

Comment:
If `scrambled.size > 1` is false, the `else` branch will crash with an `IndexOutOfBoundsException`. So we should either just remove that unnecessary size check or do proper size checking and error handling/throwing before the `if else` condition here.

Reply:
No, as it's possible that `size == 1`. As much as I dislike exceptions, I also think it's better to just let the exception be thrown here and then caught further up for handling, as handling it here is not really possible or would require more code changes.

Reply:
https://github.com/libre-tube/LibreTube/pull/7069/files#diff-bdd37ad197f5af84f6e4c0a47e640f0f0045754b709c2265e0cb0e7ae86cbacfR25 crashed if `size == 1`, or am I missing something?

Reply:
No, just me being confused by `0`-indexing. But this should indeed access the [first element and not the second one](
https://github.com/LuanRT/BgUtils/blob/54c511b2bd4e3e3707f30d2907f33167e1b61b35/src/core/challengeFetcher.ts#L46). I assume it's a bug in NewPipe, though it seems like it's never actually triggered.